fdbserver/ptxn/test/RealStorageServer.actor.cpp

```diff
@@ -0,0 +1,180 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "fdbrpc/simulator.h"
+#include "fdbserver/MockPeekCursor.h"
+#include "fdbserver/MockLogSystem.h"
+#include "fdbclient/StorageServerInterface.h"
+#include "fdbserver/IKeyValueStore.h"
+#include "fdbserver/WorkerInterface.actor.h"
+#include "fdbserver/RecoveryState.h"
+#include "fdbserver/ptxn/test/Driver.h"
+
+#include "flow/UnitTest.h"
+
+#include "flow/actorcompiler.h" // has to be last include
+
+namespace ptxn {
+
+struct ServerTestDriver {
+	ISimulator::ProcessInfo* previousProcess;
+	TaskPriority previousTask;
+
+	ACTOR static Future<Void> switchToServerProcess(ServerTestDriver* self) {
+		self->previousProcess = g_simulator.getCurrentProcess();
+		self->previousTask = g_pSimulator->getCurrentTask();
+		wait(g_simulator.onProcess(
+		    g_simulator.newProcess(
+		        "TestSystem",
+		        IPAddress(0x01010101),
+		        1,
+		        false,
+		        1,
+		        LocalityData(Optional<Standalone<StringRef>>(),
+		                     Standalone<StringRef>(deterministicRandom()->randomUniqueID().toString()),
+		                     Standalone<StringRef>(deterministicRandom()->randomUniqueID().toString()),
+		                     Optional<Standalone<StringRef>>()),
+		        ProcessClass(ProcessClass::TesterClass, ProcessClass::CommandLineSource),
+		        "",
+		        "",
+		        currentProtocolVersion),
+		    TaskPriority::DefaultYield));
+		Sim2FileSystem::newFileSystem();
+		FlowTransport::createInstance(false, 1);
+		return Void();
+	}
+
+	ACTOR static Future<Void> switchBack(ServerTestDriver* self) {
+		wait(g_simulator.onProcess(self->previousProcess, self->previousTask));
+		return Void();
+	}
+};
+
+struct StorageServerTestDriver : ServerTestDriver {
+	Reference<MockPeekCursor> mockPeekCursor;
+
+	// Default tag.
+	Tag tag = Tag(1, 1);
+	// Set by initEndpoints()
+	StorageServerInterface ssi;
+	ActorCollection actors = ActorCollection(false);
+
+	StorageServerTestDriver() : ServerTestDriver() { initEndpoints(); }
+
+	void initEndpoints() {
+		UID uid = nondeterministicRandom()->randomUniqueID();
+		ssi = StorageServerInterface(uid);
+		ssi.locality = LocalityData();
+		ssi.initEndpoints();
+	}
+
+	ACTOR static Future<Void> runStorageServer(StorageServerTestDriver* self) {
+		wait(delay(1));
+
+		StorageServerInterface& ssi = self->ssi;
+
+		KeyValueStoreType storeType = KeyValueStoreType::SSD_BTREE_V2;
```

Comment:
Please add a TODO to make this parametrizable. We'll eventually want to test this against more than the one storage server implementation.